codec/src/types/map.rs

```diff
@@ -0,0 +1,270 @@
+//! Codec implementations for HashMap.
+//!
+//! For portability and consistency between architectures,
+//! the size of the map must fit within a [`u32`].
+
+use crate::{
+    codec::{EncodeSize, Read, Write},
+    error::Error,
+    varint, Config, RangeConfig,
+};
+use bytes::{Buf, BufMut};
+use std::{collections::HashMap, hash::Hash};
+
+// Write implementation for HashMap
+impl<K: Ord + Hash + Eq + Write, V: Write> Write for HashMap<K, V> {
+    fn write(&self, buf: &mut impl BufMut) {
+        let len = u32::try_from(self.len()).expect("HashMap length exceeds u32::MAX");
+        varint::write(len, buf);
+
+        // Sort the keys to ensure deterministic encoding
+        let mut keys: Vec<_> = self.keys().collect();
```

Comment:
It may be faster to convert the whole map to a vec of `(key, value)` and `sort_by(key)`?

Reply:
...actually, not sure. That may result in moving a lot more memory around...

Reply:
disregard.